apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -589,6 +596,13 @@ type ClusterParameters struct {
 	// Other values will produce an error.
 	// +optional
 	DNSLookupFamily ClusterDNSFamilyType `json:"dnsLookupFamily,omitempty"`
+
+	// Defines the maximum requests for upstream connections. If not specified, there is no limit.
+	// see https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#envoy-v3-api-msg-config-core-v3-httpprotocoloptions
+	// for more information.
+	//
+	// +optional
+	MaxRequestsPerConnection *uint32 `json:"maxRequestsPerConnection,omitempty"`
```

Comment:
I'm curious if you considered making this configurable per-cluster, rather than having a global setting that applies to all clusters? I don't have a problem with it as-is, but could imagine some users wanting to be able to change on a per-cluster basis as well.

Reply:
Hey Skriss, 
yeah, i did think about it. 

I was thinking of something like a `AdvancedConfigPolicy` on the HTTPProxy. 

However, I wanted to run that by the design review process. Also, I want to make sure we design it with extensibility in mind as I envision all the cluster-specific config to live in that, e.g. access logs per cluster, config knobs etc.

Reply:
Also, from the discussion with Sotiris above, the `max_requests_per_connection` is a rather uncommon use case on the listener. I added it for the sake of uniformity, but if we think its not needed at this time, I am happy to revert those changes.